packages/ckeditor5-table/src/commands/insertcolumncommand.js

```diff
@@ -75,8 +75,11 @@ export default class InsertColumnCommand extends Command {
 		const referencePosition = insertBefore ? selection.getFirstPosition() : selection.getLastPosition();
 		const referenceRange = insertBefore ? selection.getFirstRange() : selection.getLastRange();
 
-		const tableCell = referenceRange.getContainedElement() || findAncestor( 'tableCell', referencePosition );
-		const table = tableCell.parent.parent;
+		const containedElement = referenceRange.getContainedElement();
+		const isTableCell = containedElement && containedElement.is( 'tableCell' );
+
+		const tableCell = isTableCell ? containedElement : findAncestor( 'tableCell', referencePosition );
+		const table = findAncestor( 'table', tableCell );
 
 		const { column } = tableUtils.getCellLocation( tableCell );
```

Comment:
I would change the whole thing to:
```js

		const affectedTableCells = getSelectionAffectedTableCells( selection );
		const columnIndexes = getColumnIndexes( affectedTableCells );

		const column = insertBefore ? columnIndexes.first : columnIndexes.last;
		const table = findAncestor( 'table', affectedTableCells[ 0 ] );
```

if the tests aren't too simplistic it works :man_shrugging:.

AFAICS the same thing can be done with inserting rows.

Reply:
The reason I would see that change is that it looks like this code tried to copy utility methods behavior that we already have. Using utility method for extracting this data makes me safe that we do it properly.